### ISSUE TITLE
Install-ChocolateyShortcut needs to handle non-existent directory structures

### ISSUE THREAD

**Issue by bc3tech** [chocolatey/choco#125]
When utilizing the `Install-ChocolateyShortcut` cmdlet, if user specifies a destination for the resulting lnk file that doesn't yet exist (eg: `start menu\<company>\<product>\<product>.lnk`) the script errors out if one or both of <company> or <product> folders don't already exist.

It's been recommended that we instead create the necessary directory structure automatically for the user, then go ahead and create the link.

**Comment by bc3tech** [comment]
In a similar vein, Install-ChocolateyShortcut isn't too useful with the current install flow of v0.9.9-rc8. This is because the Shim EXEs aren't generated until the very end, so the script errors out saying the target executable doesn't exist if the user tries to execute it with the target exe as the resulting shim. Something to think about.

**Comment by dhilgarth** [comment]
I think you shouldn't reference the shim in your shortcut. Simply reference the "real" file.

Example when using `Install-ChocolateyZipPackage`:
```
$installDir = "$(Split-Path -parent $MyInvocation.MyCommand.Definition)"
Install-ChocolateyZipPackage 'procexp' 'http://download.sysinternals.com/files/ProcessExplorer.zip' $installDir

$programs = [environment]::GetFolderPath([environment+specialfolder]::Programs)
$shortcutPath = Join-Path "$programs" "Sysinternals"
$shortcutFilePath = Join-Path $shortcutPath "Process Explorer.lnk"
if(!(Test-Path $shortcutPath)) {
    md $shortcutPath
}

Install-ChocolateyShortcut -shortcutFilePath $shortcutFilePath -targetPath (Join-Path $installDir "procexp.exe")
```


I totally agree though on the necessity to create non existent folders!

**Comment by ferventcoder** [comment]
I'm six to 1, half dozen to other on which you should reference.

**Comment by bc3tech** [comment]
Yeah if there ends up being a "best practice" toward referencing the actual file (that is what i'm doing now so my installer works) or the shim, i'm fine as long as it's actually possible ;)

**Comment by dhilgarth** [comment]
> I'm six to 1, half dozen to other on which you should reference.
Maybe it's because I am not a native speaker, but the meaning of this sentence is totally lost to me :)
Could you rephrase it please?

**Comment by gep13** [comment]
>@dhilgarth said...
Maybe it's because I am not a native speaker, but the meaning of this sentence is totally lost to me :)
Could you rephrase it please?

Basically @ferventcoder is saying that he doesn't have a preference as to which technique is followed.  For him, either way works, so pick which one makes the most sense for you.

**Comment by ferventcoder** [comment]
While I'm in fixing #122 - I might as well pick this up.

**Comment by dhilgarth** [comment]
:+1:

**Comment by bc3tech** [comment]
@ferventcoder any thought on short reference to shim vs lib\ file?

**Comment by ferventcoder** [comment]
shims can be referenced. The target doesn't have to exist yet.

**Comment by bc3tech** [comment]
Ah yes I see that as part of the commit now. nice! :+1:
